libethereum/BlockChainSync.cpp

```diff
@@ -177,6 +177,15 @@ BlockChainSync::~BlockChainSync()
 	abortSync();
 }
 
+void BlockChainSync::onBlockImported(BlockHeader const& _info)
+{
+	//if a block has been added via mining or other block import function
+	//through RPC, then we should count it as a last imported block
+	RecursiveGuard l(x_sync);
+	m_lastImportedBlock = static_cast<unsigned>(_info.number());
```

Comment:
I'm thinking now, that this could potentially break the sync.

This is called now during the sync, too, right?
`m_lastImportBlock` keeps track of the latest block we pushed into `BlockQueue` (for verification), and this can be further ahead than the blocks currently being imported into `BlockChain`, so this callback I guess can set m_lastImportedBlock to some earlier value.

So I think let's update `m_lastImportedBlock` & `m_lastImportedHash` only if the new number is greater